packages/app-store/closecom/lib/CrmService.ts

```diff
@@ -110,27 +105,76 @@ export default class CloseComCalendarService implements Calendar {
       additionalInfo: {
         customActivityTypeInstanceData,
       },
+      success: true,
     });
   }
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  async updateEvent(uid: string, event: CalendarEvent): Promise<any> {
-    return await this.closeComUpdateCustomActivity(uid, event);
+  async updateEvent(uid: string, event: CalendarEvent): Promise<CrmEvent> {
+    const updatedEvent = await this.closeComUpdateCustomActivity(uid, event);
+    return {
+      id: updatedEvent.id,
+    };
   }
 
   async deleteEvent(uid: string): Promise<void> {
-    return await this.closeComDeleteCustomActivity(uid);
+    await this.closeComDeleteCustomActivity(uid);
   }
 
-  async getAvailability(
-    _dateFrom: string,
-    _dateTo: string,
-    _selectedCalendars: IntegrationCalendar[]
-  ): Promise<EventBusyDate[]> {
-    return Promise.resolve([]);
+  async getContacts(emails: string | string[]): Promise<Contact[]> {
+    const contactsQuery = await this.closeCom.contact.search({
+      emails: Array.isArray(emails) ? emails : [emails],
+    });
+
+    return contactsQuery.data.map((contact) => {
+      return {
+        id: contact.id,
+        email: contact.emails[0].email,
+        name: contact.name,
+      };
+    });
   }
 
-  async listCalendars(_event?: CalendarEvent): Promise<IntegrationCalendar[]> {
-    return Promise.resolve([]);
+  async createContacts(contactsToCreate: ContactCreateInput[]): Promise<Contact[]> {
```

Comment:
In Close.com contacts are connected to a lead. So we need to create the lead first before we can create contacts. If there is more than one attendee in the booking, then create all contacts under the same lead.